### PR TITLE
Source Zoom: Replace JWT Auth methods with server-to-server Oauth

### LINKED ISSUE #23268: Source Zoom: JWT auth methods will be deprecated soon
## Tell us about the problem you're trying to solve

According to the document, the Zoom JWT auth methods will be deprecated on June 1, 2023.  The Zoom source connector only supports JWT and will stop working at that time. 

https://marketplace.zoom.us/docs/guides/build/jwt-app/jwt-faq/

> The JWT app type will be deprecated June 1, 2023. We recommend that you create [Server-to-Server OAuth](https://marketplace.zoom.us/docs/guides/build/server-to-server-oauth-app) or an [OAuth](https://marketplace.zoom.us/docs/guides/build/oauth-app) apps to replace the functionality of a JWT app in your account. See below for frequently asked questions about this change.

## Describe the solution you’d like
Update the Auth method

### PR DESCRIPTION
**Pull request by KqLLL**
## What
*Describe what the change is solving*
Update the Auth method
 
Resolve [Issue #23268](https://github.com/airbytehq/airbyte/issues/23268)

According to the document, the Zoom JWT auth methods will be deprecated on June 1, 2023. The Zoom source connector only supports JWT and will stop working at that time.

https://marketplace.zoom.us/docs/guides/build/jwt-app/jwt-faq/
## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Checklist
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version is set to `0.0.1`
    - [ ] `Dockerfile` has version `0.0.1`
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version has been incremented
    - [ ] Version has been bumped according to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines
    - [ ] `Dockerfile` has updated version
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` with an entry for the new version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)

- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-zoom/source_zoom/components.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+@dataclass
+class ServerToServerOauthAuthenticator(NoAuth):
+    config: Config
+    account_id: Union[InterpolatedString, str]
+    client_id: Union[InterpolatedString, str]
+    client_secret: Union[InterpolatedString, str]
+    grant_type: Union[InterpolatedString, str]
+    authorization_endpoint: Union[InterpolatedString, str]
+
+    _generate_token_time = 0
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zoom/source_zoom/components.py:28]: @KqLLL add unit test to validate the functions inside the class

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-zoom/source_zoom/components.py (hunk 1 above)
Comment (KqLLL) [reply airbyte-integrations/connectors/source-zoom/source_zoom/components.py:28]: Added unit tests

**Review by sajarin — CHANGES_REQUESTED** [review]
Thanks for this PR @KqLLL we really appreciate it. We're unable to test this PR without credentials, so we're waiting for those credentials to be generated. 

In the meantime, can you bump the versions in the Dockerfile and metadata.yaml for the Zoom connector? Please keep in mind we follow semantic versioning and this would be considered a major change.

**Comment by KqLLL** [comment]
> Thanks for this PR @KqLLL we really appreciate it. We're unable to test this PR without credentials, so we're waiting for those credentials to be generated.
> 
> In the meantime, can you bump the versions in the Dockerfile and metadata.yaml for the Zoom connector? Please keep in mind we follow semantic versioning and this would be considered a major change.

I have updated these versions in Dockerfile and metadata.yaml. 
Would you mind confirming if I'm doing this correctly?

---
Just a gentle reminder, if you take some time to study the Zoom API, you might notice that the data provided by the Zoom connector appears to be inaccurate.

**Comment by RobinHerzog** [comment]
Can please someone merge ? JWT does not exist

**Comment by marcosmarxm** [comment]
Hello, we're going to prioritize the review of this contribution next week.

**Review by marcosmarxm — COMMENTED** [review]
Hello 👋 I'm working to get Airbyte's test account to use the new Oauth method. I'm going to return the review after we get them.

**Comment by marcosmarxm** [comment]
@KqLLL still waiting to have the server-to-server Oauth enabled in our integration account. Hope this week we can have a return.

**Comment by marcosmarxm** [comment]
The internal ticket created to our infra team was moved to next sprint. Hope to get this unblock next week. @KqLLL are you available to make a call and maybe show me the connector working in your side?

**Comment by KqLLL** [comment]
> The internal ticket created to our infra team was moved to next sprint. Hope to get this unblock next week. @KqLLL are you available to make a call and maybe show me the connector working in your side?

Ok, I can show a snapshot of my connector working.

**Set in the source dashboard.**
image

**Job history.**
image

File: docs/integrations/sources/zoom.md [hunk 2]
```diff
@@ -53,11 +53,11 @@ Please [create an issue](https://github.com/airbytehq/airbyte/issues) if you see
 
 ### Requirements
 
-* Zoom JWT Token
+* Zoom Server-to-Server Oauth App
 
 ### Setup guide
 
-Please read [How to generate your JWT Token](https://marketplace.zoom.us/docs/guides/build/jwt-app).
+Please read [How to generate your Server-to-Server OAuth app ](https://developers.zoom.us/docs/internal-apps/s2s-oauth/).
```
Comment (marcosmarxm) [inline docs/integrations/sources/zoom.md:60]: @KqLLL can you add a note in Zoom documentation linking to the deprecation of JWT token?

```
:::info
JWT Token are deprecated only Server-to-Server works now.. link to Zoom
:::
```

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Can you update the documentation with the note and the Changelog with the new version?

File: airbyte-integrations/connectors/source-zoom/source_zoom/spec.yaml [hunk 3]
```diff
@@ -4,10 +4,25 @@ connectionSpecification:
   title: Zoom Spec
   type: object
   required:
-    - jwt_token
-  additionalProperties: true
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zoom/source_zoom/spec.yaml:8]: Keep `additionalProperties: true`

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments and we're good to merge the contribution. I'm still working to get credentials from our side but probably we can merge before that.

**Review by marcosmarxm — APPROVED** [review]
Thanks @KqLLL

**Comment by octavia-approvington** [comment]
Check that box <br/> TODO complete

**Review by octavia-approvington — APPROVED** [review]

**Comment by DiegoPiloni** [comment]
Hi @marcosmarxm @KqLLL ! Do you have any idea why version 1.0.0 is not in [dockerhub](https://hub.docker.com/r/airbyte/source-zoom/tags)

**Comment by jcolinger3** [comment]
> Hi @marcosmarxm @KqLLL, thanks a lot for this work ! Do you have any idea why version 1.0.0 is not in [dockerhub](https://hub.docker.com/r/airbyte/source-zoom/tags) ?

I'm wondering the same.  @DiegoPiloni did you see any change?

**Comment by DiegoPiloni** [comment]
@jcolinger3 What I understand is that https://github.com/airbytehq/airbyte/pull/30361 still needs to be approved and merged

**Pull request merged**